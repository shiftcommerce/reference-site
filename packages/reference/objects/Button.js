// Libraries
import { Component } from 'react'
import classNames from 'classnames'

class Button extends Component {
  render () {
    let {
      status,
      label,
      size,
      disabled,
      className,
      ...otherProps
    } = this.props

    disabled = disabled || false
    status = status || 'primary'
    size = size || 'lrg'

    return (
      <button
        className={classNames(className, 'o-button', `o-button--${status}`, `o-button--${size}`, {'o-button--disabled': disabled})}
        tabIndex='0'
        role='button'
        disabled={disabled}
        {...otherProps}
      >
        { label }
      </button>
    )
  }
}

export default Button
