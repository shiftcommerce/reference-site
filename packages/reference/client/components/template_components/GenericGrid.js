// Libraries
import { Component } from 'react'
import Link from 'next/link'

// Objects
import Button from '../../objects/Button'
import Image from '../../objects/Image'

class GenericGrid extends Component {
  buildSlides (component) {
    const slides = []

    for (let i of [...Array(12).keys()]) {
      const slideImage = component[`slide${i + 1}_image`]
      const slideText = component[`slide${i + 1}_text`]
      const slideLinkURL = component[`slide${i + 1}_url`]

      if (slideImage && slideText && slideLinkURL) {
        slides.push(
          <div className="c-generic-grid__card" key={i}>
            <Link href={slideLinkURL[0].canonical_path}>
              <a>
                <Image src={slideImage[0].canonical_path} />
                <p className="c-generic-grid__title">{slideText}</p>
              </a>
            </Link>
          </div>
        )
      }
    }

    return slides
  }

  render () {
    const { component } = this.props

    return (
      <section className="c-generic-grid">
        <h1 className="c-component-header">{component.header}</h1>
        <div className="c-generic-grid__container">
          {this.buildSlides(component)}
        </div>
        <Button
          className="c-generic-grid__cat-button"
          label={component.cat_text}
          status="primary"
        />
      </section>
    )
  }
}

export default GenericGrid
