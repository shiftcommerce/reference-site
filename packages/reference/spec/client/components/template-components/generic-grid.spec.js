// Components
import GenericGrid from '../../../../client/components/template-components/generic-grid'

// Objects
import Button from '../../../../client/objects/button'

// Fixtures
import genericGridData from '../../../fixtures/generic-grid'

test('GenericGrid component renders correctly', () => {
  // Arrange
  const emptyFunction = () => {}

  // Act
  const wrapper = mount(
    <GenericGrid componentData={genericGridData} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText(genericGridData.header)
  expect(wrapper).toIncludeText(genericGridData.slide1_text)
  expect(wrapper).toIncludeText(genericGridData.slide2_text)
  expect(wrapper).toIncludeText(genericGridData.slide3_text)
  expect(wrapper.find('Link').first()).toHaveProp('href', genericGridData.slide1_url[0].canonical_path)
  expect(wrapper.find('Button')).toMatchElement(<Button label={genericGridData.cat_text} onClick={emptyFunction} />)
})
