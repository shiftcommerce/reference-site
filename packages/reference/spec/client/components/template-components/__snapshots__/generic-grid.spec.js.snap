// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericGrid component renders correctly 1`] = `
<GenericGrid
  componentData={
    Object {
      "cat_text": "Cat Button",
      "cat_url": Array [
        Object {
          "canonical_path": "/homeware/dining/cat",
          "id": "/homeware/dining/cat",
          "path": "/homeware/dining/cat",
          "url": "/homeware/dining/cat",
        },
      ],
      "header": "Generic Grid",
      "slide1_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide1_text": "shop glassware",
      "slide1_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide2_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_25.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_25.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide2_text": "shop glassware2",
      "slide2_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware2",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide3_image": Array [
        Object {
          "asset_file": "1513596567.2856915-Week-44-Department-Homeware-2_26.jpg",
          "canonical_path": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_26.jpg",
          "created_at": "2017-12-18T11:29:27Z",
          "file_content_content_type": "image/jpeg",
          "file_content_filename": "Week-44-Department-Homeware-2_24.jpg",
          "file_content_size": 354806,
          "id": "157604",
          "image_height": 1483,
          "image_width": 2396,
          "reference": "26418193-2571-4bb8-943b-bb567065c4f8",
          "s3_url": "https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg",
          "updated_at": "2017-12-18T11:29:27Z",
        },
      ],
      "slide3_text": "shop glassware3",
      "slide3_url": Array [
        Object {
          "canonical_path": "/homeware/dining/glassware3",
          "id": "/homeware/dining/glassware",
          "path": "/homeware/dining/glassware",
          "url": "/homeware/dining/glassware",
        },
      ],
      "slide4_text": "",
    }
  }
>
  <section
    className="o-template-component u-center-align"
  >
    <h1
      className="c-component-header"
    >
      Generic Grid
    </h1>
    <div
      className="o-card-grid o-card-grid--3d-3m"
    >
      <div
        className="o-card-grid__card"
        key="0"
      >
        <Link
          href="/homeware/dining/glassware"
        >
          <a
            href="/homeware/dining/glassware"
            onClick={[Function]}
          >
            <Image
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_24.jpg"
            >
              <div>
                <LazyLoad
                  height={200}
                  offset={0}
                  once={false}
                  overflow={false}
                  resize={false}
                  scroll={true}
                  unmountIfInvisible={false}
                >
                  <div
                    className="lazyload-placeholder"
                    style={
                      Object {
                        "height": 200,
                      }
                    }
                  />
                </LazyLoad>
              </div>
            </Image>
            <p
              className="o-card-grid__title"
            >
              shop glassware
            </p>
          </a>
        </Link>
      </div>
      <div
        className="o-card-grid__card"
        key="1"
      >
        <Link
          href="/homeware/dining/glassware2"
        >
          <a
            href="/homeware/dining/glassware2"
            onClick={[Function]}
          >
            <Image
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_25.jpg"
            >
              <div>
                <LazyLoad
                  height={200}
                  offset={0}
                  once={false}
                  overflow={false}
                  resize={false}
                  scroll={true}
                  unmountIfInvisible={false}
                >
                  <div
                    className="lazyload-placeholder"
                    style={
                      Object {
                        "height": 200,
                      }
                    }
                  />
                </LazyLoad>
              </div>
            </Image>
            <p
              className="o-card-grid__title"
            >
              shop glassware2
            </p>
          </a>
        </Link>
      </div>
      <div
        className="o-card-grid__card"
        key="2"
      >
        <Link
          href="/homeware/dining/glassware3"
        >
          <a
            href="/homeware/dining/glassware3"
            onClick={[Function]}
          >
            <Image
              src="https://shift-platform-dev-assets.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/157604/1513596567.2856915-Week-44-Department-Homeware-2_26.jpg"
            >
              <div>
                <LazyLoad
                  height={200}
                  offset={0}
                  once={false}
                  overflow={false}
                  resize={false}
                  scroll={true}
                  unmountIfInvisible={false}
                >
                  <div
                    className="lazyload-placeholder"
                    style={
                      Object {
                        "height": 200,
                      }
                    }
                  />
                </LazyLoad>
              </div>
            </Image>
            <p
              className="o-card-grid__title"
            >
              shop glassware3
            </p>
          </a>
        </Link>
      </div>
    </div>
    <Link
      href="/homeware/dining/cat"
    >
      <a
        href="/homeware/dining/cat"
        onClick={[Function]}
      >
        <Button
          className="c-template-component__cat-button"
          label="Cat Button"
          status="primary"
        >
          <div
            className="o-button__container"
          >
            <button
              className="c-template-component__cat-button o-button o-button--primary"
              disabled={false}
              role="button"
              tabIndex="0"
            >
              Cat Button
            </button>
          </div>
        </Button>
      </a>
    </Link>
  </section>
</GenericGrid>
`;
